File: java-training/src/com/dmm/Day02/Demo8.java
package com.dmm.Day02;

class Employee {
    private String firstName;
    private String lastName;

    public Employee (String firstName, String lastName) {
        System.out.println("Constructor");
        this.firstName = firstName;
        this.lastName = lastName;
    }
    public void getFullName () {
        System.out.println(firstName + " " + lastName);
    }
}

public class Demo8 {
    public static void main(String[] args) {
        Employee employee = new Employee("Paul", "Watson");
            /*employee.firstName = "Mark";
            employee.lastName = "Watson";*/
        employee.getFullName();
    }
}

File: html-js-css-ts-training/CLASS29/typescript/demo8.ts
// class Customer {
//     firstName: string;
//     lastName: string;

//     constructor(firstName: string, lastName: string) {
//         console.log('constructor called');
//         this.firstName = firstName;
//         this.lastName = lastName;
//     }

//     getFullName() {
//         console.log(`${this.firstName} ${this.lastName}`);
//     }
// }

// let customer = new Customer ('Paul', 'Apostle');
// customer.firstName = 'Peter';
// customer.lastName = 'Apostle';
// customer.getFullName()

class Customer{
    constructor(private firstName: string, private lastName: string) {
    }

    getFullName () {
        console.log(`${this.firstName} ${this.lastName}`);
    }
}

let customer = new Customer ('Paul', 'Apostle');
customer.getFullName()